### PR TITLE
added-GlobalExceptionHandler-and-data-validation

### PR DESCRIPTION
**Pull request by margosalii**

### CODE REVIEW

File: src/main/java/mate/academy/store/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
 import java.math.BigDecimal;
 import lombok.Getter;
 import lombok.Setter;
+import mate.academy.store.validation.Isbn;
 
 @Getter
 @Setter
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class CreateBookRequestDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/store/dto/CreateBookRequestDto.java]:
```suggestion
    @NotBlank
```

check other String

File: src/main/java/mate/academy/store/controller/BookController.java [hunk 2]
```diff
@@ -34,12 +35,13 @@ public BookDto getBookById(@PathVariable Long id) {
...
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @PutMapping("/{id}")
-    public BookDto updateBook(@PathVariable Long id, @RequestBody CreateBookRequestDto requestDto) {
+    public BookDto updateBook(@PathVariable Long id,
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/store/controller/BookController.java]:
```suggestion
    public BookDto updateBook(@PathVariable @Positive Long id,
```

Add `@Validated` under the controller
Also, check other PathVariable

File: src/main/java/mate/academy/store/controller/BookController.java [hunk 3]
```diff
@@ -34,12 +35,13 @@ public BookDto getBookById(@PathVariable Long id) {
...
+                              @RequestBody @Valid CreateBookRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/store/controller/BookController.java:47]: Add `@Valid` to search(BookSearchParameters searchParameters)

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution

File: src/main/java/mate/academy/store/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,18 +1,31 @@
...
     private String title;
+
+    @NotBlank
     private String author;
+
+    @Isbn
     private String isbn;
+
+    @NotNull
+    @Min(0)
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/store/dto/CreateBookRequestDto.java]: better use @Positive

File: src/main/java/mate/academy/store/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/store/exception/CustomGlobalExceptionHandler.java]: save strings to constants

File: src/main/java/mate/academy/store/exception/CustomGlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                .stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
+    }
+
+    private String getErrorMessage(ObjectError objectError) {
+        if (objectError instanceof FieldError) {
+            String field = ((FieldError) objectError).getField();
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/store/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        if (objectError instanceof FieldError fieldError) {
            String field = fieldError.getField();
```

File: src/main/java/mate/academy/store/dto/CreateBookRequestDto.java [hunk 7]
```diff
@@ -1,18 +1,31 @@
...
     private BigDecimal price;
+
     private String description;
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/store/dto/CreateBookRequestDto.java:30]: You can validate descriptions

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**